frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java

```diff
@@ -0,0 +1,358 @@
+package com.virtuslab.gitmachete.frontend.actions.pushdialog;
+
+import java.awt.Component;
+import java.awt.event.ActionEvent;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.JRootPane;
+import javax.swing.border.Border;
+
+import com.intellij.dvcs.push.PrePushHandler;
+import com.intellij.dvcs.push.PushController;
+import com.intellij.dvcs.push.PushInfo;
+import com.intellij.dvcs.push.PushSource;
+import com.intellij.dvcs.push.PushSupport;
+import com.intellij.dvcs.push.PushTarget;
+import com.intellij.dvcs.push.VcsPushOptionValue;
+import com.intellij.dvcs.push.ui.PushLog;
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.dvcs.push.ui.VcsPushUi;
+import com.intellij.dvcs.repo.Repository;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.ui.ValidationInfo;
+import com.intellij.openapi.util.registry.Registry;
+import com.intellij.util.ui.JBDimension;
+import com.intellij.util.ui.JBUI;
+import com.intellij.util.ui.UIUtil;
+import com.intellij.util.ui.components.BorderLayoutPanel;
+import io.vavr.collection.List;
+import net.miginfocom.swing.MigLayout;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.initialization.qual.UnknownInitialization;
```

Comment:
I think this can be simplified even more now:

```diff
diff --git a/config/checker/com.intellij.astub b/config/checker/com.intellij.astub
index 3911424c..17e09df3 100644
--- a/config/checker/com.intellij.astub
+++ b/config/checker/com.intellij.astub
@@ -210,20 +210,6 @@ package com.intellij.openapi.project;
 @UIPackage package com.intellij.openapi.roots.ui.configuration.libraryEditor;
 @UIPackage package com.intellij.openapi.roots.ui.util;
 @UIPackage package com.intellij.openapi.ui;
-
-class DialogWrapper {
-  // Needed to create GitForcePushDialog and VcsPushDialogAdapterHolder "together".
-  // GitForcePushDialog must be passed (via the adapter) to the push controller
-  // that is being created within dialog's constructor.
-  JRootPane getRootPane(@UnknownInitialization DialogWrapper this);
-
-  // All of the ones below are being called from GitForcePushDialog (DialogWrapper) constructor
-  void init(@UnknownInitialization DialogWrapper this);
-  void setOKButtonText(@UnknownInitialization DialogWrapper this, String text);
-  void setOKButtonMnemonic(@UnknownInitialization DialogWrapper this, int c);
-  void setTitle(@UnknownInitialization DialogWrapper this, String title);
-}
-
 @UIPackage package com.intellij.openapi.ui.cellvalidators;
 @UIPackage package com.intellij.openapi.ui.dialogs;
 @UIPackage package com.intellij.openapi.ui.messages;
diff --git a/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java b/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java
index f8366c2a..d7d871a1 100644
--- a/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java
+++ b/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java
@@ -38,10 +38,9 @@ import com.intellij.util.ui.components.BorderLayoutPanel;
 import io.vavr.collection.List;
 import net.miginfocom.swing.MigLayout;
 import org.checkerframework.checker.guieffect.qual.UIEffect;
-import org.checkerframework.checker.initialization.qual.UnknownInitialization;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
-public class GitPushDialog extends DialogWrapper implements VcsPushUi {
+public final class GitPushDialog extends DialogWrapper implements VcsPushUi {
   private static final String DIMENSION_KEY = "Vcs.Push.Dialog.v2";
 
   private static final int CENTER_PANEL_HEIGHT = 450;
@@ -81,6 +80,8 @@ public class GitPushDialog extends DialogWrapper implements VcsPushUi {
 
     this.listPanel = pushController.getPushPanelLog();
 
+    // Note: since the class is final, `this` is already @Initialized at this point.
+
     updateOkActions();
     setOKButtonText(getPushActionName());
     setOKButtonMnemonic('P');
@@ -119,7 +120,7 @@ public class GitPushDialog extends DialogWrapper implements VcsPushUi {
 
     @Override
     @UIEffect
-    public JRootPane getRootPane(@UnknownInitialization MyVcsPushDialog this) {
+    public JRootPane getRootPane(MyVcsPushDialog this) {
       return GitPushDialog.this.getRootPane();
     }
   }
@@ -176,7 +177,7 @@ public class GitPushDialog extends DialogWrapper implements VcsPushUi {
 
   @Override
   @UIEffect
-  public JRootPane getRootPane(@UnknownInitialization GitPushDialog this) {
+  public JRootPane getRootPane(GitPushDialog this) {
     return super.getRootPane();
   }
 
@@ -206,7 +207,7 @@ public class GitPushDialog extends DialogWrapper implements VcsPushUi {
 
   @Override
   @UIEffect
-  public boolean canPush(@UnknownInitialization GitPushDialog this) {
+  public boolean canPush(GitPushDialog this) {
     return pushController != null && pushController.isPushAllowed();
   }
 
@@ -321,18 +322,18 @@ public class GitPushDialog extends DialogWrapper implements VcsPushUi {
   }
 
   @UIEffect
-  public void updateOkActions(@UnknownInitialization GitPushDialog this) {
+  public void updateOkActions(GitPushDialog this) {
     if (pushAction != null) {
       pushAction.setEnabled(canPush());
     }
   }
 
   @Override
-  public @Nullable VcsPushOptionValue getAdditionalOptionValue(@UnknownInitialization GitPushDialog this, PushSupport support) {
+  public @Nullable VcsPushOptionValue getAdditionalOptionValue(GitPushDialog this, PushSupport support) {
     return null;
   }
 
-  private String getPushActionName(@UnknownInitialization GitPushDialog this) {
+  private String getPushActionName(GitPushDialog this) {
     return isForcePushRequired ? "Force _Push" : "Push";
   }
```